Updater/EvaluateBronoldFehskeBCImpl.cpp

```diff
@@ -0,0 +1,97 @@
+#include <EvaluateBronoldFehskeBCImpl.h>
+#include <math.h>
+
+void reflectionLoop(double* fItr, double* ordinates, double* weights, double* iWeights, double* iOrdinates, double* negBasis, double* posBasis, double* vc, double* dv, double C, double elemCharge, double me, double electronAffinity, double effectiveMass, int vdim, int cdim, int numOrd, int N, int numBasis, int ignore)
+{
+  if((ignore == -1 && vc[0] < 0.0) || (ignore == 1 && vc[0] > 0.0)) {
+    for(int m=0; m<numBasis; m++) {
+      for(int n=0; n<numBasis; n++) {
+        fItr[numBasis*m + n] = 0.0;
+      }
+    }
+  }
+  else {
+    for(int m=0; m<numBasis; m++) {
+      for(int n=0; n<numBasis; n++) {
+        double Rquad[numOrd] = {};
```

Comment:
@kltrbrds there's no need to reallocate this temporary array every time through these loops. I think you can just declare the array once outside the loops and keep overwriting the elements.